### PR TITLE
feat(DS): add width to StackHorizontal component

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
For some layout, i find the need to wrap this component with a container that have a width of 100%

**What is the chosen solution to this problem?**
For now, i'm not sure it's needed to have all the sizes for the component or if just 100% is enough 🤔 

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by Benrajalu — APPROVED** [review]

**Review by sgangnant — APPROVED** [review]

**Review by tlnd-mhuchet — APPROVED** [review]

**Review by jmfrancois — COMMENTED** [review]
it feels strange to me. stack is flex, and flex size is driven by content.
I have the feeling you are trying to get a grid component using flex.

File: packages/design-system/src/components/Stack/Primitive/StackPrimitive.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const sizeOptions = {
 	XL: 'XL',
 };
 
-export const heightOptions = {
+export const sizeOptions = {
```
Comment (Nigui) [inline packages/design-system/src/components/Stack/Primitive/StackPrimitive.tsx:41]: isn't a breaking change ?

**Comment by Benrajalu** [comment]
> it feels strange to me. stack is flex, and flex size is driven by content. I have the feeling you are trying to get a grid component using flex.

Not a bad point, however from this library's standpoint, Stack is Stack. It happens to use flex as a box-model, but its main purpose is to force us into using spacing tokens for gaps in vertical or horizontal simple layouts. 

I did not provide height and width props originally because I also felt like it was the context's job. But we got several report of that decision actually discouraging the use of the Stacks. So I'd rather let it happen and have people use Stack( and thus the spacing tokens "for free" and optimizing their CSS with a lot less repetition) than keep it as a "pure" `FlexComponent`.

File: packages/design-system/src/components/Stack/StackHorizontal.stories.tsx [hunk 2]
```diff
@@ -49,7 +49,11 @@ export const manualStackArgs = {
 		defaultValue: 'nowrap',
 	},
 	height: {
-		options: Object.keys(heightOptions),
+		options: Object.keys(sizeOptions),
+		control: { type: 'select' },
+	},
+	with: {
```
Comment (Nigui) [inline packages/design-system/src/components/Stack/StackHorizontal.stories.tsx:55]: `width`

File: packages/design-system/src/components/Stack/Primitive/StackPrimitive.tsx (hunk 1 above)
Comment (romainseb) [reply packages/design-system/src/components/Stack/Primitive/StackPrimitive.tsx:41]: I don't think it's a breaking change as i dough the fact it's meant to be use in the host project 👀

Comment (Benrajalu) [reply packages/design-system/src/components/Stack/Primitive/StackPrimitive.tsx:41]: It's not exported in the lib yeah

**Comment by romainseb** [comment]
Dropped for this new one : 
https://github.com/Talend/ui/pull/4374

**Pull request closed without merging**